File: YEAR 2/SEM 1/Advanced Programming Methods/LAB 4/src/Model/Statement/CompoundStatement.java
package Model.Statement;

import Model.State.ProgramState;
import Util.Stack.MyIStack;

public class CompoundStatement implements IStatement {
    private IStatement first;
    private IStatement second;

    public CompoundStatement(IStatement first, IStatement second){
        this.first = first;
        this.second = second;
    }

    @Override
    public ProgramState execute(ProgramState programState) {
        MyIStack<IStatement> executionStack = programState.getExecutionStack();
        executionStack.push(this.second);
        executionStack.push(this.first);
        return null;
    }

    @Override
    public String toString() {
        return "(" + this.first + ";" + this.second + ")";
    }
}

File: YEAR 2/SEM 1/Advanced Programming Methods/ToyLanguage/ToyLanguage/Model/Statement/CompoundStatement.cs
using ToyLanguage.Model.State;
using ToyLanguage.Util.Stack;

namespace ToyLanguage.Model.Statement
{
    class CompoundStatement : IStatement
    {
        private readonly IStatement first;
        private readonly IStatement second;

        public CompoundStatement(IStatement first,IStatement second)
        {
            this.first = first;
            this.second = second;
        }

        public ProgramState Execute(ProgramState programState)
        {
            MyIStack<IStatement> executionStack = programState.ExecutionStack;

            executionStack.Push(second);
            executionStack.Push(first);

            return programState;
        }

        public override string ToString()
        {
            return "CompoundStatement(" + this.first + "," + this.second + ")";
        }
    }
}
